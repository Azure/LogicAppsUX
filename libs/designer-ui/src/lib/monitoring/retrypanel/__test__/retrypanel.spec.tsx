import type { RetryPanelProps } from '../index';
import { RetryPanel } from '../index';
import { setIconOptions } from '@fluentui/react';
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import { describe, vi, beforeEach, afterEach, beforeAll, afterAll, it, test, expect } from 'vitest';
import timezoneMock from 'timezone-mock';
describe('lib/monitoring/retrypanel', () => {
  let minimal: RetryPanelProps;

  beforeAll(() => {
    setIconOptions({
      disableWarnings: true,
    });
    timezoneMock.register('Etc/GMT-4');
  });

  afterAll(() => {
    timezoneMock.unregister();
  });

  beforeEach(() => {
    minimal = {
      retryHistories: [
        {
          clientRequestId: 'clientRequestId',
          code: 'code',
          startTime: '2022-02-08T19:52:00Z',
          endTime: '2022-02-10T19:52:00Z',
        },
      ],
    };
  });

  it('should render', () => {
    const tree = renderer.create(<RetryPanel {...minimal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an end time and duration when available', () => {
    const props = {
      ...minimal,
      retryHistories: [
        {
          ...minimal.retryHistories[0],
          endTime: '2022-02-08T20:10:00Z',
        },
      ],
    };

    const tree = renderer.create(<RetryPanel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a service request ID when available', () => {
    const props = {
      ...minimal,
      retryHistories: [
        {
          ...minimal.retryHistories[0],
          serviceRequestId: 'serviceRequestId',
        },
      ],
    };

    const tree = renderer.create(<RetryPanel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an error when available', () => {
    const props: RetryPanelProps = {
      ...minimal,
      retryHistories: [
        {
          ...minimal.retryHistories[0],
          error: {
            error: {
              code: 'errorCode',
              message: 'errorMessage',
            },
          },
        },
      ],
    };

    const tree = renderer.create(<RetryPanel {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('visible', () => {
    let renderer: ShallowRenderer.ShallowRenderer;

    beforeEach(() => {
      renderer = ShallowRenderer.createRenderer();
    });

    afterEach(() => {
      renderer.unmount();
    });

    it('should not render when not visible', () => {
      renderer.render(<RetryPanel {...minimal} visible={false} />);
      expect(renderer.getRenderOutput()).toBeNull();
    });
  });
});
