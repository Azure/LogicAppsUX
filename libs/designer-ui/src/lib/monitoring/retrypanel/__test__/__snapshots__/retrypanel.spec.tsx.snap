// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/monitoring/retrypanel should render 1`] = `
Array [
  <div
    className="msla-retrypanel-callout-pager"
  >
    <div
      className="msla-pager-v2"
      onClick={[Function]}
    >
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Previous"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronLeft"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="msla-pager-v2--inner"
      >
        <span
          className="css-118"
        >
          1 of 1
        </span>
      </div>
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Next"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronRight"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="msla-panel-callout-content"
  >
    <div
      className="msla-trace-inputs-outputs"
    >
      <div
        className="msla-trace-inputs-outputs-header"
      >
        <header>
          Retry
        </header>
      </div>
      <div
        className="msla-trace-values"
      >
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Duration
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Start time
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Status
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Client request ID
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
      </div>
    </div>
  </div>,
]
`;

exports[`lib/monitoring/retrypanel should render a service request ID when available 1`] = `
Array [
  <div
    className="msla-retrypanel-callout-pager"
  >
    <div
      className="msla-pager-v2"
      onClick={[Function]}
    >
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Previous"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronLeft"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="msla-pager-v2--inner"
      >
        <span
          className="css-118"
        >
          1 of 1
        </span>
      </div>
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Next"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronRight"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="msla-panel-callout-content"
  >
    <div
      className="msla-trace-inputs-outputs"
    >
      <div
        className="msla-trace-inputs-outputs-header"
      >
        <header>
          Retry
        </header>
      </div>
      <div
        className="msla-trace-values"
      >
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Duration
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Start time
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Status
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Client request ID
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Service request ID
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
      </div>
    </div>
  </div>,
]
`;

exports[`lib/monitoring/retrypanel should render an end time and duration when available 1`] = `
Array [
  <div
    className="msla-retrypanel-callout-pager"
  >
    <div
      className="msla-pager-v2"
      onClick={[Function]}
    >
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Previous"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronLeft"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="msla-pager-v2--inner"
      >
        <span
          className="css-118"
        >
          1 of 1
        </span>
      </div>
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Next"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronRight"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="msla-panel-callout-content"
  >
    <div
      className="msla-trace-inputs-outputs"
    >
      <div
        className="msla-trace-inputs-outputs-header"
      >
        <header>
          Retry
        </header>
      </div>
      <div
        className="msla-trace-values"
      >
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Duration
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Start time
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            End time
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Status
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Client request ID
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
      </div>
    </div>
  </div>,
]
`;

exports[`lib/monitoring/retrypanel should render an error when available 1`] = `
Array [
  <div
    className="msla-retrypanel-callout-pager"
  >
    <div
      className="msla-pager-v2"
      onClick={[Function]}
    >
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Previous"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronLeft"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="msla-pager-v2--inner"
      >
        <span
          className="css-118"
        >
          1 of 1
        </span>
      </div>
      <div
        className="msla-pager-failed-container"
      >
        <div
          className="ms-TooltipHost root-109"
          onBlurCapture={[Function]}
          onFocusCapture={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="none"
        >
          <button
            aria-disabled={true}
            aria-label="Next"
            className="ms-Button ms-Button--icon is-disabled root-110"
            data-is-focusable={false}
            disabled={true}
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyPress={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <span
              className="ms-Button-flexContainer flexContainer-111"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden={true}
                className="ms-Icon root-105 ms-Button-icon icon-113"
                data-icon-name="ChevronRight"
                style={
                  Object {
                    "fontFamily": undefined,
                  }
                }
              />
              <span
                className="ms-Button-textContainer textContainer-112"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="msla-panel-callout-content"
  >
    <div
      className="ms-MessageBar ms-MessageBar--severeWarning ms-MessageBar-multiline msla-request-history-panel-error root-119"
    >
      <div
        className="ms-MessageBar-content content-120"
      >
        <div
          aria-hidden={true}
          className="ms-MessageBar-icon iconContainer-121"
        >
          <i
            aria-hidden={true}
            className="icon-128"
            data-icon-name="Warning"
          />
        </div>
        <div
          aria-live="assertive"
          className="ms-MessageBar-text text-123"
          id="MessageBar32"
          role="alert"
        >
          <span
            className="ms-MessageBar-innerText innerText-124"
          />
        </div>
      </div>
    </div>
    <div
      className="msla-trace-inputs-outputs"
    >
      <div
        className="msla-trace-inputs-outputs-header"
      >
        <header>
          Retry
        </header>
      </div>
      <div
        className="msla-trace-values"
      >
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Duration
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Start time
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Status
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
        <section
          className="msla-trace-value-label"
        >
          <label
            className="msla-trace-value-display-name"
          >
            Client request ID
          </label>
          <div
            className="msla-colorizer-json-body"
          >
            <section
              style={
                Object {
                  "display": "flex",
                  "height": 19,
                  "position": "relative",
                  "textAlign": "initial",
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "100%",
                  }
                }
              >
                Loading...
              </div>
              <div
                style={
                  Object {
                    "display": "none",
                    "width": "100%",
                  }
                }
              />
            </section>
          </div>
        </section>
      </div>
    </div>
  </div>,
]
`;
